### PR TITLE
L-17 Game shop

### PR DESCRIPTION
**Pull request by IvashchenkoIvanIgorevich**

### CODE REVIEW

**Review by ZarubeyAleksandr — APPROVED** [review]

**Review by JzomitiZumatti — APPROVED** [review]

File: gameShop/src/main/java/org/example/Command.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        return Arrays.asList(SIGN_IN.key + "-" + SIGN_IN.message, SIGN_UP.key + "-" + SIGN_UP.message,
+                EXIT.key + "-" + EXIT.message);
+    }
+
+    public static List<String> getUserMenuCommands() {
+        return Arrays.asList(SHOW_ALL_USER_GAMES.key + "-" + SHOW_ALL_USER_GAMES.message,
+                SHOW_ALL_GAMES.key + "-" + SHOW_ALL_GAMES.message,
+                BUY_GAME.key + "-" + BUY_GAME.message, DEPOSIT.key + "-" + DEPOSIT.message,
+                LOGOUT.key + "-" + LOGOUT.message);
+    }
```
Comment (shaitanaleksander) [inline gameShop/src/main/java/org/example/Command.java:38]: Incorrect use of enum, enam is only for constants and you added logic to it,  Single responsibility is violated

File: gameShop/src/main/java/org/example/Executor.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.example;
+
+@FunctionalInterface
+public interface Executor {
+    void execute();
+}
```
Comment (shaitanaleksander) [inline gameShop/src/main/java/org/example/Executor.java:6]: you can use Runnable, the code needs to be removed

File: gameShop/src/main/java/org/example/Main.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class Main {
+
+    public static void main(String[] args) {
+        try(Scanner scanner = new Scanner(System.in)) {
+            scanner.useDelimiter("\n");
+            Singleton singleton = Singleton.getInstance(ConnectionSingleton.getConnection());
```
Comment (shaitanaleksander) [inline gameShop/src/main/java/org/example/Main.java:17]: Singleton is a representation of the class type, use it as a class name is bad idea

File: gameShop/src/main/java/org/example/service/Singleton.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import org.example.AppData;
+import org.example.repository.GameRepositoryImpl;
+import org.example.repository.UserRepositoryImpl;
+import org.example.repository.dao.GameRepository;
+import org.example.repository.dao.UserRepository;
+
+import java.sql.Connection;
+
+@Getter
+public class Singleton {
```
Comment (shaitanaleksander) [inline gameShop/src/main/java/org/example/service/Singleton.java:13]: class name does not represent what class is responsible for, and you probably don't want this class to be a singleton

File: gameShop/src/main/java/org/example/service/Singleton.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+        appData = AppData.getInstance();
+    }
+
+    public static Singleton getInstance(Connection connection) {
+        if (instance == null) {
+            instance = new Singleton(connection);
+        }
+        return instance;
+    }
```
Comment (shaitanaleksander) [inline gameShop/src/main/java/org/example/service/Singleton.java:38]: this part is a potential problem, what if I use this method twice with 2 different connections to different databases? Class needs to be rewritten and simplified

**Review by shaitanaleksander — APPROVED** [review]

**Pull request merged**